docs/source/user/cli.rst

```diff
@@ -0,0 +1,170 @@
+.. _cli:
+
+Command line interface
+======================
+
+.. _cli_overview:
+
+Overview
+--------
+
+Numba is a Python package, usually you ``import numba`` from Python and use the
+Python application programming interface (API). However, Numba also ships with a
+command line interface (CLI), i.e. a tool ``numba`` that you gets installed when
+you install Numba.
+
+Currently, the only purpose of the CLI is to allow you to quickly show some
+information about your system and installation, or to quickly get some debugging
+information for a Python script using Numba. To see which options are available,
+use ``numba --help``:
+
+::
+
+    $ numba --help
+    usage: numba [-h] [--annotate] [--dump-llvm] [--dump-optimized]
+                 [--dump-assembly] [--dump-cfg] [--dump-ast]
+                 [--annotate-html ANNOTATE_HTML] [-s]
+                 [filename]
+
+    positional arguments:
+      filename              Python source filename
+
+    optional arguments:
+      -h, --help            show this help message and exit
+      --annotate            Annotate source
+      --dump-llvm           Print generated llvm assembly
+      --dump-optimized      Dump the optimized llvm assembly
+      --dump-assembly       Dump the LLVM generated assembly
+      --dump-cfg            [Deprecated] Dump the control flow graph
+      --dump-ast            [Deprecated] Dump the AST
+      --annotate-html ANNOTATE_HTML
+                            Output source annotation as html
+      -s, --sysinfo         Output system information for bug reporting
+
+Sometimes it can happen that you get a "command not found" error when you type
+``numba``, because your ``PATH`` isn't configured properly. In that case you can
+use the equivalent commmand ``python -m numba``. If that still gives "command
+not found", try to ``import numba`` as suggested here:
+:ref:`numba-source-install-check`.
+
+.. _cli_sysinfo:
+
+System information
+------------------
+
+The ``numba --sysinfo`` (or ``numba -s`` for short) command prints a lot of
+information about your system and your Numba installation and relevant
+dependencies.
+
+You can execute it from the terminal like this::
+
+    $ numba -s
+    $ python -m numba -s
+
+To see your system information from IPython or Jupyter, prefix the command with
+an exclamation mark like this::
+
+    !numba -s
+    !python -m numba -s
+
+The two versions ``numba`` and ``python -m numba`` are the same. The first is
```

Comment:
I also feel like there is some duplication in here, but at the same time this information might be useful for someone who is fairly new to the Python world. I am a bit undecided because I don't have a good suggestion for this so I would tend to leave it as is.

Reply:
O.K. so after some more thinking, here is my proposed solution. I would mention, that `numba` has an equivalent form as `python -m numba` at the very beginning of this text. I.e. before listing the output of `numba --help` verbatim. I would mention that:

* There is an alternative `python -m numba`
* This accesses the same functionality as the `numba` command
* List the circumstances where it can be used.
* The different way to invoke this (cli, jupyter)

Right now, it feel like this information is 'bolted' on to other section of the documentation and I think it would be good to have all this in one place. Does that sound reasonable?